src/NetMQ/OutgoingSocketExtensions.cs

```diff
@@ -98,8 +200,182 @@ public static IOutgoingSocket SendMore([NotNull] this IOutgoingSocket socket, [N
 
         #endregion
 
+        #endregion
+
+        #region Sending a multipart message as byte arrays
+
+        #region Blocking
+
+        public static void SendMultipartBytes([NotNull] this IOutgoingSocket socket, params byte[][] frames)
+        {
+            SendMultipartBytes(socket, (IEnumerable<byte[]>)frames);
+        }
+
+        public static void SendMultipartBytes([NotNull] this IOutgoingSocket socket, IEnumerable<byte[]> frames)
+        {
+            var enumerator = frames.GetEnumerator();
```

Comment:
C# `foreach` disposes the IEnumerator if it implements IDisposable. I'm not sure it's important, but as this API is public it's theoretically possible client code could depend upon this behaviour.

```csharp
E enumerator = (collection).GetEnumerator();
try {
   while (enumerator.MoveNext()) {
      ElementType element = (ElementType)enumerator.Current;
      statement;
   }
}
finally {
   IDisposable disposable = enumerator as System.IDisposable;
   if (disposable != null) disposable.Dispose();
}
```